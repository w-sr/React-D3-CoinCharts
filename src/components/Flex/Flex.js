import styled from "styled-components";

// export type Props = {
//   auto?: boolean,
//   column?: boolean,
//   reverse?: boolean,
//   justify?: JustifyValue,
//   align?: AlignValue,
//   wrap?: WrapValue,
//   className?: string,
//   center?: boolean,
//   onClick?: Function,
// };

const Flex = styled.div`
  display: flex;
  ${props => (props.onClick ? "cursor: pointer;" : "")};
  ${props => (props.auto ? "flex: 1 1 auto;" : "")};
  ${props => (props.center ? `justify-content: center; align-items: center;` : ``)};
  ${props => (props.justify ? `justify-content: ${props.justify};` : "")};
  ${props => (props.align ? `align-items: ${props.align};` : "")};
  ${props => (props.wrap ? `flex-wrap: ${props.wrap};` : "")};
  flex-direction: ${({ column, reverse }) => {
    const postFix = reverse ? "-reverse" : "";
    return column ? `column${postFix}` : `row${postFix}`;
  }};
`;

Flex.displayName = "Flex";

export default Flex;
