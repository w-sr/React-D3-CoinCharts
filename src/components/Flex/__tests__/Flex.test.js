import React from "react";
import { render } from "react-testing-library";

import Flex from "../";

describe("<Flex />", () => {
  it("renders without crashing", () => {
    const { container } = render(<Flex />);
    expect(container.firstChild).not.toBeNull();
    expect(container.firstChild).toMatchSnapshot();
  });
});
