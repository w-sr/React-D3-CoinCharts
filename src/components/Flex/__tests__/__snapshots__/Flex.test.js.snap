// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Flex /> renders without crashing 1`] = `
<div
  class="sc-bdVaJa cAcgKX"
/>
`;
